KEYCLOAK-13757 update JDG version to 8.1 - testsuite updates

tkyjovsk opened the pull request.

tkyjovsk committed:
KEYCLOAK-13757 update JDG version to 8.1 - testsuite updates

mposolda commented:
@tkyjovsk Do you please have a link to the pipeline run with cluster and cross-dc tests?

mposolda commented:
@tkyjovsk Thanks for the update! Few notes:

- It will be nice to update HOW-TO-RUN.md and add some info into the cross-dc section about how to run the cross-dc tests with the new and old versions of the infinispan server. IMO the new infinispan server is more important and hence the instructions for it should be default (added to the top) when the instructions for "legacy" server would be better somewhere at the bottom

- The GH actions should ideally be updated to use the new infinispan server IMO. The option is to use both new and old, but IMO that will be unecessary overhead and will add more error-proneness. So my vote is to keep just new infinispan in the GH actions

- I guess pipeline should be updated, so that cross-dc tests use both new and old infinispan server? Also I am missing the successful pipeline run with this PR?

pdrozd commented:
DG 8.x should be the primary for upstream and also for GH actions. 7.x is just for the current stream. The PR to the pipeline is required.

pskopek commented:
The pipeline PR #8 has been merged.

vramik reviewed (changes requested):
Thank you @tkyjovsk for the PR. Overall changes looks good to me. 

I found new modules a little bit confusing: `cache-server/infinispan/infinispan`, `cache-server/infinispan/datagrid`, `cache-server/jboss/infinispan`, `cache-server/jboss/datagrid` .. then I saw `CacheStatisticsControllerEnricher.isOldInfinispan()` method which bringed more light into it. Do we still need to test infinispan 9 or lower in upstream? 

IMO it'd be worth to document it somewhere, probably HOW-TO-RUN.md would be correct place as @mposolda suggested.

Please provide links to pipeline runs.

tkyjovsk commented on testsuite/integration-arquillian/servers/auth-server/jboss/pom.xml:
```diff
@@ -697,6 +701,13 @@
         </profile>
+        
+        <profile>
+            <id>cache-auth</id>
```
@vramik The scenario is for testing authenticated access to Infinispan/Datagrid caches via the HotRod protocol.

vramik replied:
That makes sense, thank you.

tkyjovsk commented on testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/containers/InfinispanServerDeployableContainer.java:
```diff
@@ -0,0 +1,230 @@
+package org.keycloak.testsuite.arquillian.containers;
```
Addressed since `d66706e`.

tkyjovsk commented on testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/containers/InfinispanServerConfiguration.java:
```diff
@@ -0,0 +1,85 @@
+package org.keycloak.testsuite.arquillian.containers;
```
Addressed since `d66706e`.

tkyjovsk commented:
In revision `d66706e` I did some refactoring.
- I renamed `cache-server-jboss-infinispan` to `cache-server-legacy-infinispan` (and `-datagrid` modules respectively).
- I updated the HOW-TO doc with some additional info.

Known issues:
- the `auth-servers-crossdc-undertow` profile currently doesn't work
- the `cache-auth` profile doesn't currently work with the legacy Infinispan/Datagrid modules

I propose to fix these issues separately afterwards.

tkyjovsk committed:
KEYCLOAK-13757 fix for KEYCLOAK-18267_KEYCLOAK-17254

vramik reviewed (dismissed):
Thank you @tkyjovsk! The PR LGTM.

mposolda reviewed (changes requested):
@tkyjovsk PR looks good to me. Thanks! Just one note: Is it possible to update HOW-TO-RUN.md with some notes and TODOs for the configurations and combinations, which doesn't work as you mentioned in the comment above? Ideally if you create JIRA for those issues and mention the JIRA numbers directly in HOW-TO-RUN (So that in the future, the reader of HOW-TO-RUN can evaluate if JIRA was already fixed etc).

This update in HOW-TO-RUN fortunately doesn't need another pipeline run, so hope it is not a big issue :-)

Besides that, I am leaving to @pdrozd approve before merge this PR.

iankko commented on testsuite/integration-arquillian/HOW-TO-RUN.md:
```diff
@@ -795,17 +795,34 @@ land by adjusting load balancer configuration (e.g. to direct the traffic to onl
+    -DgroupId=com.redhat -DartifactId=datagrid -Dpackaging=zip -Dclassifier=bin -DgeneratePom=true \
+    -Dversion=${DATAGRID_VERSION} -Dfile=redhat-datagrid-${DATAGRID_VERSION}-server.zip
+
+For Data Grid 7 and older the use: `-Dfile=jboss-datagrid-${DATAGRID_VERSION}-server.zip`.
```
Just nitpick, I would drop `the` here. IOW so it reads: `For Data Grid 7 and older use: ...`

tkyjovsk replied:
Addressed in `40d2230`.

iankko reviewed (dismissed):
LGMT AFAICT (based just on code review, didn't manually test the change, but since the referenced pipeline run returns zero failures, assuming it works OK)

tkyjovsk commented:
@mposolda Addressed your comments in revision `40d2230`.

mposolda reviewed (dismissed):
@tkyjovsk Thanks!

tkyjovsk commented:
Revision `6c753ce` includes [commit](https://github.com/tkyjovsk/keycloak/commit/884084f7b713a0de8364df646434b490e537daeb) which is needed after https://github.com/keycloak/keycloak/pull/8106 was merged.
